bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderWatcherHandler.java

```diff
@@ -0,0 +1,291 @@
+
+                    if (diff < config.diffHours) {
+                        aFile.setName(dirToList + "/" + currentFileName);
+                        dirList.add("ftp:/" + ftpClient.getRemoteAddress().toString() + aFile.getName());
+                    }
+                }
+            }
+        }
+        return dirList;
+    }
+
+    protected Runnable periodicConnectRunnable = new Runnable() {
+        int reply;
+
+        @Override
+        public void run() {
+            if (ftp == null || ftp.isConnected() == false) {
+                if (config.connectionTimeout <= 0) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Connetction timeout can't be negative");
+                    return;
+                }
+
+                if (config.secureMode.equals("NONE")) {
+                    ftp = new FTPClient();
+                } else {
+                    FTPSClient ftps;
+                    if (config.secureMode.equals("IMPLICIT")) {
+                        ftps = new FTPSClient(true);
+                    } else if (config.secureMode.equals("EXPLICIT")) {
+                        ftps = new FTPSClient(false);
+                    } else {
+                        ftps = null;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Unsupported secure mode");
+                        return;
+                    }
+                    ftp = ftps;
+                }
+
+                reply = 0;
+                ftp.setListHiddenFiles(config.listHidden);
+                ftp.setConnectTimeout(config.connectionTimeout * 1000);
+                // Uncomment to see FTP response
+                // ftp.addProtocolCommandListener(new PrintCommandListener(new PrintWriter(System.out), true));
+
+                try {
+                    if (config.ftpPort > 0) {
+                        ftp.connect(config.ftpAddress, config.ftpPort);
+                    } else {
+                        ftp.connect(config.ftpAddress);
+                    }
+                    reply = ftp.getReplyCode();
+
+                    if (!FTPReply.isPositiveCompletion(reply)) {
+                        ftp.disconnect();
+                        logger.debug("FTP server refused connection.");
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "FTP server refused connection.");
+                        return;
+                    }
+                } catch (IOException e) {
+                    if (ftp.isConnected()) {
+                        try {
+                            ftp.disconnect();
+                        } catch (IOException e2) {
+                            logger.debug("Can't disconect: ", e2);
+                        }
+                    }
+                    logger.debug("Can't conect: ", e);
```

Comment:
Not needed or is the full stack trace providing additional information? Thing status changes are logged anyway.